extensions/kafka/src/test/java/com/hazelcast/jet/kafka/impl/StreamKafkaPTest.java

```diff
@@ -316,6 +316,42 @@ public void when_snapshotSaved_then_offsetsRestored() throws Exception {
         assertNoMoreItems(processor, outbox);
     }
 
+    @Test
+    public void when_duplicateTopicsProvide_then_uniqueTopicsSubscribed() {
+        HazelcastInstance[] instances = instances();
+        assertClusterSizeEventually(2, instances);
+
+        // need new topic because we want 2 partitions only
+        String topic = randomString();
+        kafkaTestSupport.createTopic(topic, 2);
+
+        Pipeline p = Pipeline.create();
+        // Pass the same topic twice
+        p.readFrom(KafkaSources.kafka(properties(), topic, topic))
+         .withoutTimestamps()
+         .setLocalParallelism(1)
+         .writeTo(Sinks.list("sink"));
+
+        JobConfig config = new JobConfig();
+        Job job = instances[0].getJet().newJob(p, config);
+
+        assertJobStatusEventually(job, JobStatus.RUNNING, 10);
+
+        int messageCount = 1000;
+        for (int i = 0; i < messageCount; i++) {
+            kafkaTestSupport.produce(topic, i, Integer.toString(i));
+        }
+
+        IList<Object> list = instances[0].getList("sink");
+        try {
+            assertTrueEventually(() -> assertThat(list).hasSize(messageCount), 15);
+            sleepSeconds(1);
+            assertThat(list).hasSize(messageCount);
```

Comment:
Here, we may use `assertTrueAllTheTime` method:  https://github.com/hazelcast/hazelcast/blob/f34dba56c3349cb3c825bdebd2bc50bf99d28118/hazelcast/src/test/java/com/hazelcast/test/HazelcastTestSupport.java#L1176-L1185